lib/transports/file/index.js

```diff
@@ -78,9 +80,42 @@ function archiveLog(stream) {
     stream.end();
   }
 
+  var logFileName = stream.path.substring(stream.path.lastIndexOf('/') + 1);
+  var folderPath = stream.path.substring(0, stream.path.lastIndexOf('/'));
+
+  // Ensure that there are only 2 old log files maximum at this moment.
+  try {
+    var allFiles = fs.readdirSync(folderPath);
+    var postfixes = [];
+    var converted = null;
+    var i = 0;
+
+    for (i = 0; i < allFiles.length; i++) {
+      if (allFiles[i].includes(logFileName + '-')) {
+        converted = parseInt(allFiles[i].substring(logFileName.length + 1), 10);
+        if (isNaN(converted)) {
+          converted = 0;
+        }
+        postfixes.push(converted);
+      }
+    }
+    
+    postfixes.sort(function(a, b) { return a - b; });
+
+    for (i = 0; i < postfixes.length - (transport.maxItems - 2); i++) {
```

Comment:
`(transport.maxItems - 2)` On this way you will get just 2 old logs instead requested 3 old log!